rust/automerge/src/query/list_state.rs

```diff
@@ -0,0 +1,142 @@
+use crate::op_tree::{LastInsert, OpTreeNode};
+use crate::query::QueryResult;
+use crate::types::{Key, ListEncoding, Op};
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct ListState {
+    pub(crate) encoding: ListEncoding,
+    pub(crate) last_seen: Option<Key>,
+    pub(crate) last_width: usize,
+    pub(crate) target: usize,
+    pub(crate) index: usize,
+    pub(crate) pos: usize,
+    pub(crate) clean: bool,
```

Comment:
As per comments on the `query::Index::clean` field, I think this might be easier to read if it was named `has_seen_insert` or similar.